datadog_lambda/tracing.py

```diff
@@ -411,6 +411,9 @@ def is_legacy_lambda_step_function(event):
     """
     Check if the event is a step function that called a legacy lambda
     """
+    if not isinstance(event, dict):
+        return False
+
     event = event.get("Payload", {})
```

Comment:
This isn't super related to this incident, but I think it's worth changing anyway. I'm wondering if we can exit early if `"Payload"` is not in the event. By then we already know that it's not gonna be a step function event. So, exiting early would prevent the allocation of the `{}` and all the searching on the next line.

Reply:
Good callout 👍🏽 , will add an early exit